src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -490,13 +510,21 @@ def _check_lvm_reject_reasons(self):
             # only check generic if no vgs are present. Vgs might hold lvs and
             # that might cause 'locked' to trigger
             rejected.extend(self._check_generic_reject_reasons())
+        if not self.is_acceptable_device_generic:
+            rejected.append('Device type is not acceptable. It should be raw device or partition')
 
         return len(rejected) == 0, rejected
 
-    def _check_raw_reject_reasons(self):
+    def _check_raw_reject_reasons(self, compat):
         rejected = self._check_generic_reject_reasons()
```

Comment:
You want to get rid of this here afaiu. Because you want to accept additional devices that the generic part would reject, drop this call and implement the checks you do want yourself (like size, ceph-disk and such).